hw04-generics/src/main/java/ru/otus/homework/CustomerReverseOrder.java

```diff
@@ -0,0 +1,16 @@
+package ru.otus.homework;
+
+import java.util.ArrayDeque;
+
+public class CustomerReverseOrder {
+
+    private final ArrayDeque<Customer> customers = new ArrayDeque<>();
```

Comment:
```suggestion
    private final Deque<Customer> customers = new ArrayDeque<>();
```

Тип объявлять лучше интерфейсом.